### DISCUSSION TITLE
Maintain variables in EnvironmentFile

### DISCUSSION THREAD

**Discussion by vishnubraj** [nautobot/nautobot#588] [Q&A]
We can maintain the variables in the "/etc/default/nautobot" and call the variable in the systemctl files using the "EnvironmentFile" field, This will help maintain the variables properly. 

content of /etc/default/nautobot

```
# cat /etc/default/nautobot
NAUTOBOT_ROOT=/opt/nautobot
NAUTOBOT_ALLOWED_HOSTS=*
NAUTOBOT_DB_USER=nautobot
NAUTOBOT_DB_PASSWORD=test123
```

content of systemd file
```
[Unit]
Description=Nautobot Request Queue Worker
Documentation=https://nautobot.readthedocs.io/en/stable/
After=network-online.target
Wants=network-online.target

[Service]
Type=simple
Environment="NAUTOBOT_ROOT=/opt/nautobot"
EnvironmentFile=/etc/default/nautobot
User=nautobot
Group=nautobot
WorkingDirectory=/opt/nautobot

ExecStart=/opt/nautobot/bin/nautobot-server rqworker

Restart=on-failure
RestartSec=30
PrivateTmp=true

[Install]
WantedBy=multi-user.target
```

Please update the below documentation for the same. 

https://nautobot.readthedocs.io/en/latest/installation/wsgi/

**Comment by jathanism** [comment]
Thanks @vishnubraj for the suggestion! This seems like a useful addendum to our configuration/deployment documentation for users who wish to deploy Nautobot using the environment variables without needing to make modifications to the default configuration.

I observed you still have `NAUTOBOT_ROOT` defined in the unit file, and also in `/etc/default/nautobot`. Could you please confirm that defining all environment variables inside of `/etc/default/nautobot` works as expected? I suspect yes, but just wanted to be explicit!

Thank you!

**Comment by smk4664** [comment]
I can confirm that having NAUTOBOT_ROOT in the EnvironmentFile would mean you don't need it in the Environment=...  Also, the EnvironmentFile can live anywhere that the Service has access to.

**Comment by smk4664** [comment]
The /etc/default is used for sysvinit (init.d) startup scripts to default environment variables see https://help.ubuntu.com/community/UbuntuBootupHowto for more information. 

Systemd can take any path for these files (as long as it can access them). https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile=

**Comment by jedelman8** [comment]
@vishnubraj  You may have seen the issue be closed, but just a heads up converted this to a discussion to get more input and feedback.  Feel free to provide a write up here so others can take advantage of your deployment methodology.
